base/src/main/kotlin/net/corda/v5/base/types/LayeredPropertyMap.kt

```diff
@@ -7,57 +7,102 @@ package net.corda.v5.base.types
  * The layered property map provides simple conversions to a possibly complex objects which can use several keys in
  * dot-notation. Take as an example of the map:
  *
- * "corda.name" to "CN=me, O=R3, L=Dublin, C=Ireland",
- * "corda.sessionKey" to "ABCDEF...",
- * "corda.endpoints.0.url" to "localhost",
- * "corda.endpoints.0.protocolVersion" to "1",
- * "corda.endpoints.1.url" to "localhost",
- * "corda.endpoints.1.protocolVersion" to "2"
+ * - "corda.name" to "CN=me, O=R3, L=Dublin, C=Ireland",
+ * - "corda.sessionKey" to "ABCDEF...",
+ * - "corda.endpoints.0.url" to "localhost",
+ * - "corda.endpoints.0.protocolVersion" to "1",
+ * - "corda.endpoints.1.url" to "localhost",
+ * - "corda.endpoints.1.protocolVersion" to "2"
  *
  * That map can be parsed into:
  * - MemberX500Name using parse("corda.name", MemberX500Name::class.java)
  * - session PublicKey using parse("corda.sessionKey", PublicKey::class.java)
  * - list of endpoints using parseList("corda.endpoints", EndpointInfo::class.java)
  *
+ * Example usages:
+ *
+ * ```java
+ * Set<Map.Entry<String, String>> entries = propertyMap.getEntries();
+ * String groupId = propertyMap.parse("corda.groupId", String.class);
+ * Instant modifiedTime = propertyMap.parseOrNull("corda.modifiedTime", Instant.class);
+ * Set<String> additionalInformation = propertyMap.parseSet("additional.names", String.class);
+ * List<EndpointInfo> endpoints = propertyMap.parseList("corda.endpoints", EndpointInfo.class);
+ * ```
+ *
+ * ```kotlin
+ * val entries = propertyMap.entries
+ * val groupId = propertyMap.parse("corda.groupId", String::class.java)
+ * val modifiedTime = propertyMap.parseOrNull("corda.modifiedTime", Instant::class.java)
+ * val additionalInformation = propertyMap.parseSet("additional.names", String::class.java)
+ * val endpoints = propertyMap.parseList("corda.endpoints", EndpointInfo::class.java)
+ * ```
+ *
  * The default implementation of the [LayeredPropertyMap] is extendable by supplying implementations of custom
  * converters using OSGi. Out of box it supports conversion to simple types like Int, Boolean,
- * as well as MemberX500Name.
+ * as well as [MemberX500Name].
+ *
+ * @property entries Returns [Set] of all entries in the underlying map.
+ *
+ * @author Alexey Kadyrov
+ * @since DP2
```

Comment:
Not sure that we are using  @since either

Reply:
Same comment as above.

Reply:
Here as well, some of the APIs are from C4 some new and DP2 of which version, C5? As there is no real guidance on how the value should be determined juts don't set until that is clarified.

@nadinequinr3 there must be clear guidance on how to set that tag which is currently doesn't exists. I would be of the opinion that we need it but probably it's premature ATM

Reply:
@mattbradburyr3 Do we want use "Corda 5 DP2" in @since for everything? Does it matter if it was modified from Corda 4 code?